draft-ietf-taps-impl.md

```diff
@@ -166,14 +166,11 @@ It is expected that the database of system policies and the method of looking up
 
 The process of establishing a network connection begins when an application expresses intent to communicate with a Remote Endpoint by calling Initiate. (At this point, any constraints or requirements the application may have on the connection are available from pre-establishment.) The process can be considered complete once there is at least one Protocol Stack that has completed any required setup to the point that it can transmit and receive the application's data.
 
-Connection establishment is divided into two top-level steps: Candidate Gathering, to identify the paths, protocols, and endpoints to use, and Candidate Racing (see Section 4.2.2 of {{I-D.ietf-taps-arch}}), in which the necessary protocol handshakes are conducted so that the transport system can select which set to use.
+Connection establishment is divided into two top-level steps: Candidate Gathering, to identify the potential paths, protocols, and endpoints to use, and Candidate Racing (see Section 4.2.2 of {{I-D.ietf-taps-arch}}), in which the necessary protocol handshakes are conducted so that the transport system can select which set to use.
 
-This document structures the candidates for racing as a tree as terminological convention. While a 
-a tree structure is not the only way in which racing can be implemented, it does ease the illustration of how racing works. 
+In its most basic form, the connection-establishment process might just involve identifying a single IP address (and port) to which the application wishes to connect, and starting a TCP handshake to establish a stream to that IP address, using the system's current default path (i.e., by using the current default network interface). However, each step may also differ depending on the requirements for the connection: if, instead of a known IP address, the endpoint is identified by a hostname, then there may be a choice between multiple resolved addresses; some protocols may not need any transport handshake to establish communication (such as UDP), while others may require several layers of handshakes, such as TLS over TCP; finally, there may also be multiple paths available (i.e., via additional network interfaces besides the system default);
```

Comment:
Nit: I think "connection-establishment" should just be "connection establishment".

More broadly, I don't love how this makes the default seem like connecting to an IP address — "if, instead of a known IP address...". The default is going to be higher-level endpoints, and that's a core tenet of the API.

Reply:
Agreed. Perhaps replacing the start of that sentence with something like "more commonly, when the endpoint..." will solve this.

Reply:
We could change

> if, instead of a known IP address, the endpoint is identified by a hostname, then there may be a choice between multiple resolved addresses;

to simply state that

> an endpoint is more commonly identified by a hostname which may resolve into more than one IP address;

Reply:
I think Tommy and Anna wanted to emphasize that having a name resolution step is the more common case.

Reply:
I see, I'm a little unsure this specific change is really needed... and it doesn't seem to read well, but if it is needed then please consider  /which may/that can/ ... and /resolve into/resolve to/